pkg/resources/stream_on_directory_table.go

```diff
@@ -137,7 +139,8 @@ func UpdateStreamOnDirectoryTable(ctx context.Context, d *schema.ResourceData, m
 	}
 
 	// change on these fields can not be ForceNew because then the object is dropped explicitly and copying grants does not have effect
-	if keys := changedKeys(d, "stage"); len(keys) > 0 || isStale(d) {
+	// recreate when the stream is stale - see https://community.snowflake.com/s/article/using-tasks-to-avoid-stale-streams-when-incoming-data-is-empty
+	if keys := changedKeys(d, "stage", "stale"); len(keys) > 0 {
```

Comment:
why we use changed keys and not the HasChange/GetChange? Also, what if this is changed here from true to false - which would happen after the recreation, right?

Reply:
Well, we could use `HasChanges`, but now we can log which fields were exactly changed, which is useful information for potential debugging, IMO. 

I added a comm to RecreateWhenStreamIsStale. Basically, our desired state is `false` for this field. That's why when we detect `true`, we set a new value to `false` to detect this change and recreate the stream. But after the recreation the old value would be `false`.

Reply:
But we get this info from the plan, right? I asked because this is different than the logic in other resources and I think we should have a reason for inconsistencies. Yes, logging is nice, but would we benefit also from it in other resources or is it just applicable here? If we could have leverage this for better logging in every resource then I would say that this is a discussion topic for the unified logging later on.

Reply:
And thanks for the true -> false explanation.

Reply:
Yes, we have this info in the plan, but the plan doesn't say that the resource will be recreated, and why that happens - there may be changes in other fields that don't cause recreation. That's why I added this logging only in resources with copy grants.